agent/api/task/service_connect.go

```diff
@@ -72,3 +86,33 @@ type DNSConfigEntry struct {
 	HostName string `json:"hostName"`
 	Address  string `json:"address"`
 }
+
+// ParseServiceConnectAttachment parses the service connect container name and service connect config value
+// from the given attachment.
+func ParseServiceConnectAttachment(scAttachment *ecsacs.Attachment, networkMode string, ipv6Enabled bool) (*ServiceConnectConfig, error) {
```

Comment:
Given that the this class also has the actual logic of setting up, could you please move this logic to a new class - `ServiceConnectAttachmentParser` ?

Reply:
Moved the function to a new file in the commit 3d8d95406e74fc25d3c00d384a11cd5cd4ae55f9. Please take a look on agent/api/task/service_connect_attachment_parser.go. Thanks.